### PR TITLE
Bump to new releases and make site names unique per test

### PR DESCRIPTION
**Pull request by rfay**
## The Problem:

* We upgraded to site repos without the Name: field in config.yaml - bump to new releases and test
* Now apps can have unique names instead of all having the Name: in the upstream test repo. Use unique names.

## The Fix:

* Add the Name: provided by the test. This is done in testcommon.Prepare()
* Note that because this means testcommon.Prepare() now uses functions from ddevapp, the ddevapp config_test.go encountered circular imports. The workaround was to make config_test.go be in ddevapp_test package (instead of ddevapp) for "black box testing" that avoid the circular import. [Lesser-known features of go test](https://splice.com/blog/lesser-known-features-go-test/) has a nice paragraph on "The black box test package"

## The Test:

Test should run successfully; you should see properly named apps created (and then destroyed)

### CODE REVIEW

File: pkg/testcommon/testcommon.go [hunk 1]
```diff
@@ -67,6 +69,23 @@ func (site *TestSite) Prepare() error {
 		site.Cleanup()
 	}
 
+	// If our test site has a Name: then update the config file to reflect that.
+	if site.Name != "" {
```
Comment (tannerjfco) [inline pkg/testcommon/testcommon.go]: @rfay Could this possibly be handled by opening the existing config file of the test site and appending the name to it? It seems like that would address this need without having an import cycle between testcommon and ddevapp.

Comment (rfay) [inline pkg/testcommon/testcommon.go]: Thanks for the idea. I hate to use nonstandard methods to access the config file, but that may be the ticket.

**Comment by rfay** [comment]
This wasn't supposed to be so hard. The adding of names and then resolving the import cycles made it more work, but I learned a lot I hope. I think this is ready for review.

File: pkg/plugins/platform/local.go [hunk 2]
```diff
@@ -498,3 +498,8 @@ func (l *LocalApp) AddHostsEntry() error {
 	err = system.RunCommandPipe("sudo", hostnameArgs)
 	return err
 }
+
+// constructContainerName builds a container name given the type (web/db/dba) and the appName
+func constructContainerName(containerType string, appName string) string {
```
Comment (beeradb) [inline pkg/plugins/platform/local.go]: I'd really like to see us use `FindContainerByLabels()` here and then `util.ContainerName()` on the result.

This hasn't been the case to date, but I would really like to see us be deliberate and consistent about the way we interact with containers.

Comment (beeradb) [inline pkg/plugins/platform/local.go]: In all places you are using it I believe you could also just use `app.FindContainerByType("web")` or `app.FindContainerByType("db")` which uses `FindContainerByLabels()` under the hood.

Comment (rfay) [inline pkg/plugins/platform/local.go]: Nice. This is actually just a simplification of the inline sprintf that was there previously. Sounds like your approach would be excellent.

**Pull request merged**